pkg/proxy/util/utils.go

```diff
@@ -241,8 +241,5 @@ func AppendPortIfNeeded(addr string, port int32) string {
 	}
 
 	// Append port to address.
-	if ip.To4() != nil {
-		return fmt.Sprintf("%s:%d", addr, port)
-	}
-	return fmt.Sprintf("[%s]:%d", addr, port)
+	return net.JoinHostPort(addr, string(port))
```

Comment:
do you know if we have to change `pkg/proxy/apis/config/v1alpha1/defaults.go` to not return the ipv6 default address with square brackets?

Reply:
Look like this line is failing (produces a garbled port)